WalletWasabi/WabiSabi/Client/CoinRefrigerator.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using WalletWasabi.Blockchain.TransactionOutputs;
+
+namespace WalletWasabi.WabiSabi.Client
+{
+	public class CoinRefrigerator
+	{
+		private Dictionary<SmartCoin, DateTimeOffset> CoinsInFridge { get; } = new();
+		private TimeSpan FreezeTime { get; } = TimeSpan.FromSeconds(90);
+
+		public void Freeze(IEnumerable<SmartCoin> coins)
+		{
+			foreach (var coin in coins)
+			{
+				if (!CoinsInFridge.ContainsKey(coin))
+				{
+					CoinsInFridge.TryAdd(coin, DateTimeOffset.UtcNow);
+				}
+				else
+				{
+					CoinsInFridge[coin] = DateTimeOffset.UtcNow;
+				}
```

Comment:
I believe one can just do:

```suggestion
				CoinsInFridge[coin] = DateTimeOffset.UtcNow;
```

Reply:
Magnificent!